File: solutions/kotlin/word-count/1/src/main/kotlin/WordCount.kt
object WordCount {
    fun phrase(s: String): Map<String, Int> = s.split(Regex("[^a-zA-Z0-9']"))
        .filterNot { it.isEmpty() }
        .map { it.toLowerCase().trim('\'') }
        .fold(mutableMapOf()) {
            acc, word -> acc[word] = acc[word]?.plus(1) ?: 1; acc
        }
}

File: solutions/typescript/word-count/1/word-count.ts
export function count(sentence: string): Map<string, number> {
  const words = sentence.trim().toLowerCase().split(/[\t\n ]+/)
  const occurences = new Map<string, number>()

  words?.forEach((word) => {
    if (occurences.has(word)) {
      occurences.set(word, occurences.get(word)!! + 1)
    } else {
      occurences.set(word, 1)
    }
  })

  return occurences
}
